### PR TITLE
Bugfix: don't reply to names that are too long.

### LINKED ISSUE #67: wildcard can be added even if the name is too long
this gets a reply:
dig @localhost -p 1053 east._._.production.skydns.local SRV 

so does this:
dig @localhost -p 1053 _.east._.*.production.skydns.local SRV

and this:
dig @localhost -p 1053 _._.east._._.production.skydns.local SRV 

And think the last two should not be answer, but instead give back NXDOMAIN

### PR DESCRIPTION
**Pull request by miekg**
This closes #67

### CODE REVIEW

File: registry/registry.go [hunk 1]
```diff
@@ -152,9 +152,11 @@ func (r *DefaultRegistry) Get(domain string) ([]msg.Service, error) {
 	}
 
 	tree := dns.SplitDomainName(domain)
-
+	if len(tree) > 4 {
```
Comment (miekg) [inline registry/registry.go]: I couldn't figure out of what or if there was a need for queries beyond this limit. The 'len(tree) < 6' below suggest there is...

File: registry/registry.go [hunk 2]
```diff
@@ -152,9 +152,11 @@ func (r *DefaultRegistry) Get(domain string) ([]msg.Service, error) {
...
+		return nil, ErrNotExists
+	}
 	// Domains can be partial, and we should assume wildcards for the unsupplied portions
-	if len(tree) < 6 {
+	if len(tree) < 6 {	// TODO(miek): why 6 here?
```
Comment (miekg) [inline registry/registry.go]: @erikstmartin can you elaborate on the 6 here? I'm also in favor of making this a named constant.

Comment (erikstmartin) [inline registry/registry.go]: I missed this inquiry, the 6 represents each of the identifying portions of the domain:

<uuid>.<host>.<region>.<version>.<service>.<environment>

**Comment by bketelsen** [comment]
What's the status on this? Is it needed?

**Comment by miekg** [comment]
This is still be needed, but longer names can just be nxdomained. Something
this does not happen now iirc

**Comment by miekg** [comment]
Ok, I've clean up this PR, rebased it to the latest, added some docs and most important removed my < 4 which was clearly wrong.

**Pull request closed without merging**